contracts/random/RandomSeedProvider.sol

```diff
@@ -0,0 +1,292 @@
+    /// @dev random seeds that have been produced.
+    mapping(uint256 requestId => bytes32 randomValue) public randomOutput;
+
+    /// @notice The index of the next seed value to be produced.
+    uint256 public nextRandomIndex;
+
+    /// @notice The block number in which the last seed value was generated.
+    uint256 public lastBlockRandomGenerated;
+
+    /// @notice The block when the last offchain random request occurred.
+    /// @dev This is used to limit off-chain random requests to once per block.
+    uint256 private lastBlockOffchainRequest;
+
+    /// @notice The request id returned in the previous off-chain random request.
+    /// @dev This is used to limit off-chain random requests to once per block.
+    uint256 private prevOffchainRandomRequest;
+
+    /// @notice The source of new random numbers. This could be the special values for
+    /// @notice TRADITIONAL or RANDAO or the address of a Offchain Random Source contract.
+    /// @dev This value is return with the request ids. This allows off-chain random sources
+    /// @dev to be switched without stopping in-flight random values from being retrieved.
+    address public randomSource;
+
+    /// @notice Indicates that this blockchain supports the PREVRANDAO opcode and that
+    /// @notice PREVRANDAO should be used rather than block.hash for on-chain random values.
+    bool public ranDaoAvailable;
+
+    /// @notice Indicates an address is allow listed for the offchain random provider.
+    /// @dev Having an allow list prevents spammers from requesting one random number per block,
+    /// @dev thus incurring cost on Immutable for no benefit.
+    mapping(address gameContract => bool approved) public approvedForOffchainRandom;
+
+    // @notice The version of the storage layout.
+    // @dev This storage slot will be used during upgrades.
+    uint256 public version;
+
+    /**
+     * @notice Initialize the contract for use with a transparent proxy.
+     * @param _roleAdmin is the account that can add and remove addresses that have
+     *        RANDOM_ADMIN_ROLE privilege.
+     * @param _randomAdmin is the account that has RANDOM_ADMIN_ROLE privilege.
+     * @param _upgradeAdmin is the account that has UPGRADE_ADMIN_ROLE privilege.
+     * @param _ranDaoAvailable indicates if the chain supports the PRERANDAO opcode.
+     */
+    function initialize(
+        address _roleAdmin,
+        address _randomAdmin,
+        address _upgradeAdmin,
+        bool _ranDaoAvailable
+    ) public virtual initializer {
+        _grantRole(DEFAULT_ADMIN_ROLE, _roleAdmin);
+        _grantRole(RANDOM_ADMIN_ROLE, _randomAdmin);
+        _grantRole(UPGRADE_ADMIN_ROLE, _upgradeAdmin);
+
+        // Generate an initial "random" seed.
+        // Use the chain id as an input into the random number generator to ensure
+        // all random numbers are personalised to this chain.
+        randomOutput[0] = keccak256(abi.encodePacked(block.chainid, block.number));
```

Comment:
out of curiosity is there a reason this wouldn't use the `blockhash`  in place of `block.number` similar to the [approach used when generating subsequent on-chain random values](https://github.com/immutable/contracts/pull/166/files#diff-f0d90b866fb87d83db9110a94fe29ec00c3c5f4733a86c0ffd74a45a5022e0eeR281)?

Reply:
For this situation, either will be good enough. My reasoning for block.number was to have a value that combined with chain id would be different from any other chain. If you were certain that chain id was unique, you could just use it. You could argue that more entropy will be derived from block.hash. The block hash indirectly includes the block number in its hash, so is probably better. 
I will switch this to block.hash.